experimental/Schemes/CoherentSheaves.jl

```diff
@@ -0,0 +1,461 @@
+  OOX::StructureSheafOfRings
+  M::PreSheafType
+
+  function HomSheaf(F::AbsCoherentSheaf, G::AbsCoherentSheaf)
+    X = scheme(F)
+    X === scheme(G) || error("sheaves must be defined over the same scheme")
+    OOX = sheaf_of_rings(F)
+    OOX === sheaf_of_rings(G) || error("sheaves must be defined over the same sheaves of rings")
+
+    ### Checks for open containment.
+    # Copy from the sheaves of Modules
+    function is_open_func(U::PrincipalOpenSubset, V::PrincipalOpenSubset)
+      C = default_covering(X)
+      A = ambient_scheme(U)
+      A in C || return false
+      B = ambient_scheme(V)
+      B in C || return false
+      if A === B
+        is_subset(U, V) || return false
+      else
+        GG = C[A, B] # Get the glueing
+        f, g = glueing_morphisms(GG)
+        is_subset(U, domain(f)) || return false
+        is_subset(V, domain(g)) || return false
+        gU = preimage(g, U)
+        is_subset(gU, V) || return false
+      end
+      return true
+    end
+    function is_open_func(U::PrincipalOpenSubset, Y::AbsCoveredScheme)
+      return Y === X && ambient_scheme(U) in default_covering(X)
+    end
+    function is_open_func(U::AbsSpec, Y::AbsCoveredScheme)
+      return Y === X && U in default_covering(X)
+    end
+    function is_open_func(Z::AbsCoveredScheme, Y::AbsCoveredScheme)
+      return X === Y === Z
+    end
+    function is_open_func(U::AbsSpec, V::AbsSpec)
+      U in default_covering(X) || return false
+      V in default_covering(X) || return false
+      GG = default_covering(X)[U, V]
+      return issubset(U, glueing_domains(GG)[1])
+    end
+    function is_open_func(U::PrincipalOpenSubset, V::AbsSpec)
+      V in default_covering(X) || return false
+      ambient_scheme(U) === V && return true
+      W = ambient_scheme(U)
+      W in default_covering(X) || return false
+      GG = default_covering(X)[W, V]
+      return is_subset(U, glueing_domains(GG)[1])
+    end
+
+    ### Production of the modules on open sets; to be cached
+    function production_func(U::AbsSpec)
+      return hom(F(U), G(U))[1]
+    end
+
+    function restriction_func(V::AbsSpec, MV::ModuleFP, U::AbsSpec, MU::ModuleFP)
+      dom_res = F(V, U)
+      cod_res = G(V, U)
+      f = gens(F(V))
+      rf = dom_res.(f)
+      # The following two lines will work, because a set of generators for ℱ(V) 
+      # always restricts to a set of generators for ℱ(U). Due to changes of 
+      # charts, this might be a non-trivial change of bases, however.
+      dom_sub, inc = sub(F(U), rf)
+      B = [coordinates(e, dom_sub) for e in ambient_representatives_generators(F(U))]
+      images = elem_type(MU)[]
+      for phi in gens(MV)
+        phi_map = element_to_homomorphism(phi)
+        images_f = [sum([B[i][j]*cod_res(phi_map(f[j])) for j in 1:length(f)]) for i in 1:length(B)]
+        psi = hom(F(U), G(U), images_f)
+        push!(images, homomorphism_to_element(MU, psi))
+      end
```

Comment:
@jankoboehm : I implemented the restriction of homomorphisms to open subsets here. But I'm not sure, I am using your machinery as intended. In particular the conversion of elements to homomorphisms and back seems to be a bit tedious. Do you provide something like "induced hom from left" and "induced hom from right" for "composition" of Hom-modules with homomorphisms to the domain and from the codomain?

Reply:
The construction from what we have is not so difficult and too many shortcuts might in the end be difficult to remember and confusing, but I think it is a good idea to introduce it to provide all the automatisms of Hom modules. We have lift_homomorphism_contravariant and lift_homomorphism_contravariant, though not exactly what you want.

Reply:
Ok, I'll have a look. But then you say that my solution is in the scope of how you intended users to handle the modules? 

If not, feel free to propose another solution here!